jerry-core/ecma/operations/ecma-typedarray-object.c

```diff
@@ -1727,6 +1738,83 @@ ecma_op_create_typedarray (const ecma_value_t *arguments_list_p, /**< the arg li
         }
       }
     }
+    else if (ecma_object_class_is (obj_p, ECMA_OBJECT_CLASS_SHARED_ARRAY_BUFFER))
+    {
+      /* 22.2.1.5 */
+      ecma_object_t *shared_arraybuffer_p = obj_p;
+      ecma_value_t arg2 = ((arguments_list_len > 1) ? arguments_list_p[1]
+                                                    : ECMA_VALUE_UNDEFINED);
+
+      ecma_value_t arg3 = ((arguments_list_len > 2) ? arguments_list_p[2]
+                                                    : ECMA_VALUE_UNDEFINED);
+
+      ecma_number_t offset;
+      if (ECMA_IS_VALUE_ERROR (ecma_op_to_index (arg2, &offset)))
+      {
+        return ECMA_VALUE_ERROR;
+      }
+
+      if (ecma_number_is_negative (offset) && ecma_number_is_zero (offset))
+      {
+        offset = 0;
+      }
+      if (ecma_number_is_negative (offset) || fmod (offset, (1 << element_size_shift)) != 0)
```

Comment:
This is hard to read. A newline should be added before this `if`. Or is it an `else if`?